src/client/websocket/handlers/TYPING_START.js

```diff
@@ -4,8 +4,23 @@ const { Events } = require('../../../util/Constants');
 const textBasedChannelTypes = ['dm', 'text', 'news'];
 
 module.exports = (client, { d: data }) => {
-  const channel = client.channels.cache.get(data.channel_id);
-  const user = client.users.cache.get(data.user_id);
+  let channel = client.channels.cache.get(data.channel_id);
+  let user = client.channels.cache.get(data.user_id);
+
+  if (!channel || !user) {
+    client.channels.fetch(data.channel_id).then(rChannel => {
+      channel = rChannel;
+      client.users.fetch(data.user_id).then(rUser => {
+        user = rUser;
+        triggerEvent(channel, user, data, client);
+      });
+    });
```

Comment:
although this PR will probably be denied,
- you don't handle any errors here
- you assign the resolved value of both promises to another value, when you could have used async/await to re-assign channel and user, or just not re-assigned them

both of these are bad practice, and inconsistent with other behaviour throughout the lib